### DISCUSSION TITLE
forge fundme- lesson14, getting error with prank

### DISCUSSION THREAD

**Discussion by heheboii11** [Cyfrin/foundry-full-course-cu#1325] [Q&A]
function testmultiplefunderswithdraw() public {
        uint160 funders = 10;
        uint160 index = 2;
        for (uint160 i = index; i < funders; i++) {
            hoax(address(i), send_value);
            fundme.fund{value: send_value};
        }
        uint256 initialfundmebalance = address(fundme).balance;
        uint256 initialownerbalance = fundme.getOwner().balance;
        vm.startPrank(fundme.getOwner());
        fundme.withdraw();
        vm.stopPrank();

        // uint256 endingfundmebalance = address(fundme).balance;
        // uint256 endingownerbalance = fundme.getOwner().balance;
        assertEq(initialfundmebalance, 0);
        assert(
            initialfundmebalance + initialownerbalance ==
                fundme.getOwner().balance
        );
    }

error:[FAIL. Reason: cannot overwrite a prank until it is applied at least once] testmultiplefunderswithdraw() (gas: 4439)
<img width="1710" alt="Screenshot 2024-01-14 at 7 15 41 PM" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/132984428/29684123-1c9a-4972-8b6c-1008f54851d2">

**Comment by tobezhanabi** [comment]
it is `startBroadcast();` and` stopBroadcast();` not startPrank or stopPrank

**Reply by heheboii11** [reply]
no bro it is not startbroadcast because we are here pranking as owner so it should be vm.startPrank(); and vm.stopPrank();
I even tried as you said it wont work . but dont know what is wrong here but i am stuck here

**Reply by allwin199** [reply]
Can you post your entire test file? It would be easy to help

**Comment by heheboii11** [comment]
// SPDX-License-Identifier: MIT
pragma solidity ^0.8.18;

import {Test} from "forge-std/Test.sol";
import {FundMe} from "../src/fundme.sol";
import {console} from "forge-std/console.sol";
import {DeployFundme} from "../script/DeployFundme.s.sol";

contract FundmeTest is Test {
    FundMe fundme;
    address user =
        address(uint160(uint256(keccak256(abi.encodePacked("user")))));
    uint256 constant send_value = 10 ether;
    uint256 constant starting_value = 10 ether;

    function setUp() external {
        DeployFundme deployfundme = new DeployFundme();
        fundme = deployfundme.run();
        vm.deal(user, starting_value);
    }

    modifier funded() {
        vm.prank(user);
        fundme.fund{value: send_value}();
        _;
    }

    function testFundme() public {
        assertEq(fundme.MINIMUM_USD(), 5e18);
    }

    function testowner() public {
        console.log(address(this));
        console.log(fundme.getOwner());
        assertEq(fundme.getOwner(), msg.sender);
    }

    function testversion() public {
        uint256 version = fundme.getVersion();

        console.log(version);

        assertEq(version, 4);
    }

    function testamountfunded() public funded {
        console.log(address(this));
        assertEq(fundme.getaddressfundamount(user), send_value);
    }

    function testaddressfunded() public funded {
        assertEq(fundme.getaddressfunded(0), user);
    }

    function testwithdrawnotownerfailure() public {
        console.log(address(this));
        vm.prank(user);
        vm.expectRevert();
        fundme.withdraw();
    }

    function testSingleWithdraw() public {
        uint256 initialfundmebalance = address(fundme).balance;
        uint256 initialownerbalance = fundme.getOwner().balance;
        vm.prank(fundme.getOwner());
        fundme.withdraw();
        uint256 endingfundmebalance = address(fundme).balance;
        uint256 endingownerbalance = fundme.getOwner().balance;
        assertEq(endingfundmebalance, 0);
        assertEq(
            initialfundmebalance + initialownerbalance,
            endingownerbalance
        );
    }

    function testmultiplefunderswithdraw() public {
        uint160 funders = 10;
        uint160 index = 2;
        for (uint160 i = index; i < funders; i++) {
            hoax(address(i), send_value);
            fundme.fund{value: send_value};
        }
        uint256 initialfundmebalance = address(fundme).balance;
        uint256 initialownerbalance = fundme.getOwner().balance;
        vm.startPrank(fundme.getOwner());
        fundme.withdraw();
        vm.stopPrank();

        // uint256 endingfundmebalance = address(fundme).balance;
        // uint256 endingownerbalance = fundme.getOwner().balance;
        assert(initialfundmebalance == 0);
        assert(
            initialfundmebalance + initialownerbalance ==
                fundme.getOwner().balance
        );
    }
}
 this is the total test file @allwin199

**Reply by allwin199** [reply]
in funded modifier instead of doing `vm.prank()` try `vm.startPrank()` and `vm.stopPrank()`
Even in other places try `vm.startPrank()` and `vm.stopPrank()`

**Comment by heheboii11** [comment]
still getting the same error

**Reply by allwin199** [reply]
clone the actual repo and try running the tests. Then compare it with your code and check for errors.

**Reply by heheboii11** [reply]
i just used the code from patricks repo it just worked but i cannot find any difference between his and mine

**Comment by Sushobhan03** [comment]
Uncomment the 'endingFundMeBalance' and then replace the line, 
``` solidity
assertEq(initialfundmebalance == 0)
```
with

```solidity
assertEq(endingfundmebalance == 0)
```
Let me know, if it works.

**Comment by SSFoxrr** [comment]
I know this is older but I had the same issue and the problem was using "hoax".

Instead use the following:

```
        for(uint160 i = startingFunderIndex; i < numberOfFunders; i++){
            vm.prank(fundMe.getOwner());
            fundMe.withdraw();
        }
```

I understand there is likely a way to use hoax and it be correct, but this is effectively the same as using the hoax included code. It will give you the same outcome.

Let me know if this helps and select my comment as the correct answer to your question.

All the best.
